elyra/pipeline/processor_kfp.py

```diff
@@ -173,13 +175,23 @@ def process(self, pipeline):
                 else:
                     raise lve
 
-            # Create a new experiment. If it already exists this is
-            # a no-op.
-            experiment = client.create_experiment(name=experiment_name,
-                                                  namespace=user_namespace)
-            self.log_pipeline_info(pipeline_name,
-                                   f'Created experiment {experiment_name}',
-                                   duration=(time.time() - t0_all))
+            try:
+                # Create a new experiment. If it already exists this is
+                # a no-op.
+                experiment = client.create_experiment(name=experiment_name,
+                                                      namespace=user_namespace)
+                self.log_pipeline_info(pipeline_name,
+                                       f'Created experiment {experiment_name}',
+                                       duration=(time.time() - t0_all))
+            except ApiException as ae:
+                self.log.error(f'Could not create experiment {experiment_name}: {ae.reason} ({ae.status})')
+                if ae.body:
```

Comment:
Based on the `init` function for the `ApiException` class, there is a chance that the body attribute of the exception would be `None`, although I was unable to reproduce with a scenario of my own to test it. Not sure if this `if/else` is superfluous in that case or if I should keep it.

Reply:
This is fine.  What I will sometimes do for these kinds of things is use the `if  block` to format the additional detail - with the empty string being its initial value - then only have one `raise` just outside the `if block` - kinda like...
```python
additional_info = ""
if ae.body:
    error_msg = json.loads(ae.body)
    additional_info = f": {error_msg['error']}"

raise RuntimeError(f'Could not create experiment {experiment_name}: {ae.reason} ({ae.status})' +
                                       f'{additional_info}') from ae
```